src/tree/StepTreeItem.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as moment from 'moment';
+import * as prettyMs from 'pretty-ms';
+import { AzureTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
+import { ActionWorkflowStepData, Conclusion, Status } from '../gitHubTypings';
+import { convertConclusionToVerb, convertStatusToVerb } from '../utils/gitHubUtils';
+import { treeUtils } from "../utils/treeUtils";
+import { IAzureResourceTreeItem } from './IAzureResourceTreeItem';
+import { JobTreeItem } from './JobTreeItem';
+
+export class StepTreeItem extends AzureTreeItem implements IAzureResourceTreeItem {
+
+    public static contextValue: string = 'azureStaticStep';
+    public readonly contextValue: string = StepTreeItem.contextValue;
+    public parent: JobTreeItem;
+    public data: ActionWorkflowStepData;
+
+    constructor(parent: JobTreeItem, data: ActionWorkflowStepData) {
+        super(parent);
+        this.data = data;
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return treeUtils.getActionIconPath(<Status>this.data.status, <Conclusion>this.data.conclusion);
+    }
+
+    public get id(): string {
+        return `${this.parent.id}/${this.data.name}`;
+    }
+
+    public get name(): string {
+        return this.data.name;
+    }
+
+    public get label(): string {
+        return this.name;
+    }
+
+    public get description(): string {
+        if (this.data.conclusion !== null) {
+            const elapsedMs: number = this.completedDate.getTime() - this.startedDate.getTime();
+            return `${convertConclusionToVerb(<Conclusion>this.data.conclusion)} in ${prettyMs(elapsedMs)}`;
+        } else {
+            return `${convertStatusToVerb(<Status>this.data.status)} ${this.startedDate.getTime() === 0 ? '' : moment(this.startedDate).fromNow()}`;
```

Comment:
What does fromNow() return if the startedDate is 0 (ie now)?  

I feel like instead of having a blank, saying queued now is more similar to what GitHub does anyway.

Reply:
This check is making sure the `this.startedDate` isn't the epoch (which is what it is if github hasn't given us a started it yet. 

From now when the startedDate when `startedDate.getTime() === 0` returns like 31 years ago or something since that's the epoch.

Reply:
Ohh, like the beginning of the ms clock?  I didn't realize that's what it was called.  Okay, gotcha.